crates/bevy_ecs/src/entity/map_entities.rs

```diff
@@ -70,6 +70,55 @@ pub struct EntityMap {
     map: HashMap<Entity, Entity>,
 }
 
+pub struct EntityMapper<'m> {
+    map: &'m mut EntityMap,
+    dead_start: Entity,
+    generations: u32,
+}
+
+impl<'m> EntityMapper<'m> {
+    /// Reserves the allocated dead entities within the world.
+    pub fn save(mut self, world: &mut World) {
+        if self.generations == 0 {
+            return;
+        }
```

Comment:
Leaving a comment for myself here as I noticed this when going over feedback - this short-circuit forgets to despawn the entity